src/lib/transaction_database/intf.ml

```diff
@@ -1,19 +1,32 @@
 open Coda_base
 open Signature_lib
 
+(** Transaction database is a database that stores transactions for a client.
+    It would query a set of transactions involving a peer with a certain public
+    key. The main purpose for this database is for a developer to store
+    transactions and perform pagination queries of transactions via GraphQL *)
 module type S = sig
   type t
 
   type time
 
-  val create : ?directory_name:string -> unit -> t
+  val create : Logger.t -> string -> t
 
   val close : t -> unit
 
   val add : t -> Transaction.t -> time -> unit
 
+  (* get_transactions t pk would query all the transactions involving pk as a
```

Comment:
can you use `(**` comments?

Reply:
for each of these in the signature here

Reply:
nit: Instead of using the `would query`, use `queries` (present tense) for all sentences and all the comments here.